### PR TITLE
WIP: Implement 'pool': An attribute & constraint that lets you split a cluster

### PR DESCRIPTION
**Pull request by EvanKrall**
...into multiple parts, without having to use Mesos's roles.

### CODE REVIEW

**Comment by EvanKrall** [comment]
I've marked this as WIP because I want to be very careful about merging and rolling this out. As is, this will cause a large bounce.

File: paasta_tools/utils.py [hunk 1]
```diff
@@ -218,6 +218,18 @@ def get_extra_volumes(self):
         <https://mesosphere.github.io/marathon/docs/native-docker.html>`_"""
         return self.config_dict.get('extra_volumes', [])
 
+    def get_pool(self):
+        """Which pool of nodes this job should run on. This can be used to mitigate noisy neighbors, by putting
```
Comment (solarkennedy) [inline paasta_tools/utils.py]: Can you move these docs, or some of them to the actual docs location?

**Comment by solarkennedy** [comment]
Are you sure you don't want to do the thing where if pool is not None, add the constraint?

Also of course we'll need to add the attributes to our slaves now

**Comment by EvanKrall** [comment]
I'm not sure what you mean in your first sentence. My default behavior is pretty clearly specified: a constraint of `['pool', 'LIKE', 'default']`.

I've got an internal review for adding the attribute to our slaves.

**Comment by solarkennedy** [comment]
Gotcha, it sound like you are going to deploy the new attribute first, and then optionally add this constraint to a bunch of apps, then push this code?

**Comment by EvanKrall** [comment]
Yeah, that's the plan.

**Pull request merged**